src/librustc_metadata/locator.rs

```diff
@@ -328,8 +328,9 @@ impl<'a> CrateLocator<'a> {
             crate_name,
             exact_paths: if hash.is_none() {
                 sess.opts.externs.get(&crate_name.as_str()).into_iter()
-                    .flat_map(|entry| entry.locations.iter())
-                    .filter_map(|location| location.clone().map(PathBuf::from)).collect()
+                    .filter_map(|entry| entry.files())
```

Comment:
Hmm, since `get` returns an `Option`, I suspect the `.into_iter().filter_map()` pair can be replaced with `and_then` or something.

Reply:
I don't see how.  It needs to translate `Option<&ExternEntry>` to `Vec<PathBuf>`. Somewhere it needs to unwrap and flatten.  I can't think of any other way to write it that doesn't end up adding several more method calls (`map_or_else`, `unwrap_or_else`, etc.).